packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -534,6 +534,76 @@ class _ZoomExitTransitionState extends State<_ZoomExitTransition> with _ZoomTran
   }
 }
 
+// This transition slides a new page in from right to left while fading it in,
+// and simultaneously slides the previous page out to the left while fading it out.
+// This transition is designed to match the Android U activity transition.
+class _FadeForwardsPageTransition extends StatelessWidget {
+  const _FadeForwardsPageTransition({
+    required this.animation,
+    required this.secondaryAnimation,
+    this.backgroundColor,
+    this.child,
+  });
+
+  final Animation<double> animation;
+
+  final Animation<double> secondaryAnimation;
+
+  final Color? backgroundColor;
+
+  final Widget? child;
+
+  // The new page slides in from right to left.
+  static final Animatable<Offset> _forwardTranslationTween = Tween<Offset>(
+    begin: const Offset(0.25, 0.0),
+    end: Offset.zero,
+  ).chain(CurveTween(curve: FadeForwardsPageTransitionsBuilder._transitionCurve));
+
+  // The old page slides back from left to right.
+  static final Animatable<Offset> _backwardTranslationTween = Tween<Offset>(
+    begin: Offset.zero,
+    end: const Offset(0.25, 0.0),
+  ).chain(CurveTween(curve: FadeForwardsPageTransitionsBuilder._transitionCurve));
+
+  @override
+  Widget build(BuildContext context) {
+    return DualTransitionBuilder(
+      animation: animation,
+      forwardBuilder: (
+        BuildContext context,
+        Animation<double> animation,
+        Widget? child
+      ) {
+        return FadeTransition(
+          opacity: FadeForwardsPageTransitionsBuilder._fadeInTransition.animate(animation),
+          child: SlideTransition(
+            position: _forwardTranslationTween.animate(animation),
+            child: child,
+          ),
+        );
+      },
+      reverseBuilder: (
+        BuildContext context,
+        Animation<double> animation,
+        Widget? child
+      ) {
+        return FadeTransition(
+          opacity: FadeForwardsPageTransitionsBuilder._fadeOutTransition.animate(animation),
+          child: SlideTransition(
+            position: _backwardTranslationTween.animate(animation),
+            child: child,
+          ),
+        );
+      },
+      child: FadeForwardsPageTransitionsBuilder._delegatedTransition(
```

Comment:
The delegatedTransition won't be passed if the route below uses the same delegatedTransition. So setting a default secondary transition is still necessary, and using the delegatedTransition limits code duplication.

It's not passed because some transitions, mainly the old Zoom transition and the Cupertino transition, get messed up in specific cases if they get overwritten by the same transition.

Reply:
This seems a bit weird. So the issue is that due to code dedup, delegateTransition is also used as secondary transition for itself. and because overriding the same animation can cause some weird situations so we have check to make sure the route above won't override the bottom if delegateTransition is the same.

What about cases where a route doesn't use its delegateTransition as its secondary transition? in that case won't the delegateTransition of the route above get ignored?

Reply:
> What about cases where a route doesn't use its delegateTransition as its secondary transition? in that case won't the delegateTransition of the route above get ignored?

No, it wouldn't get ignored. If a pages' default secondary transition is different from it's delegatedTransition, which can happen, you can set a unique transition for both. But if they are the same, you may as well use the delegatedTransition in the place of the secondary transition.

Reply:
talked offline, this may potentially be an issue, but it is tangential to this pr.